nntrainer/layers/permute_layer.cpp

```diff
@@ -36,7 +36,7 @@ const std::string PermuteLayer::type = "permute";
  * @return const std::string string to return
  */
 static std::string
-buildTrasposeString(const std::array<props::Direction, 3> &arr) {
+buildTrasposeString(const std::array<props::PermuteDims, 3> &arr) {
   std::stringstream ss;
   ss << arr[0].get() << ':' << arr[1].get() << ':' << arr[2].get();
```

Comment:
not directly related to this PR, but consider first getting unique values from this array and then applying the transpose operation. Also, the current implementation expects exactly 3 dimensions to be set for permute, but even two dimensions should be enough to use the permute layer.

Reply:
You mean applying array directly? Yes, this should better be supported.

Since we have fixed number of dimension(4), I am concerning that the exact semantics of having two dimension, If this is solid, I think we can apply for two dimension.

Reply:
Specifying 3 dimensions for transposing 2 dimensions is an inconvenience. I will approve for this, this can be considered as TODO. Also, there needs to a way to ensure that all the values are unique.

Reply:
> But, there needs to a way to ensure that all the values are unique

Could you check out if this is what you've stated? https://github.com/nnstreamer/nntrainer/blob/6cf6dd2844b4fd6bb49b2728ab06af2432ed60e4/nntrainer/layers/permute_layer.cpp#L47

> Specifying 3 dimensions for transposing 2 dimensions is an inconvenience.

afaik, for other frameworks it is forced that `rank(shape)` `len(direction)` must match, as we have semantics that `rank(shape) == 3`, I think we need stronger assumption or semantics to make it clear.

For example, given ​shape `2:1:4
if 2:1 is given, possible interpretations are
  1. `1 <-> 4`, because h is 1 when starting from channel
  1. invalid, because height 1 means there is height does not count.
  1. invalid, because `len(direction) == 2` must start from zero

if 0:1 is given, possible interpretations are
  1. `2 <-> 1` starting from channel
  1. `1 <-> 4` starting from height (but invalid because channel is > 2)
  1.  `2 <-> 4` starting from channel but height is 1 so it is not counted.

If we have solid interpretation considering we have fixed number of dimensions I think we could support num 2.

Reply:
> Could you check out if this is what you've stated?

Thanks for the link. I missed this. For later, you may consider moving such constraints to setProperty().

Agreed on the part that 2 dimensions might lead to confusion. 
However, this raises question (outside the scope of this PR), dimension `0:1:2` maps to `c:h:w` and in some cases, `0:1:2:3` maps to `b:c:h:w`. Let's make it consistent.

Reply:
>  this raises question

Yes, for now, we have firm semantics it is always `c:h:w` and current behavior (ignoring batch dimension) coherent with keras `PermuteLayer`. but `tf.transpose` will start from batch. 


Please throw me ideas if any.

My current thinking is transpose in the near future, so we might be needing two semantics (with different name I guess :p)

Or we can consider it as one semantic, i guess, if we never allow putting batch dimension inside *.ini or property.

Thank you for raising the issue :)

Reply:
https://www.tensorflow.org/api_docs/python/tf/keras/layers/Permute
tf.keras.layers.permute starts dimensions with 1 (I think they left 0 for batch). 
Let's follow these semantics to avoid confusion.

Reply:
Thanks, made a change to accept 1~3